spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -169,6 +171,16 @@ public void setTransactionIdPrefix(String transactionIdPrefix) {
 		this.transactionIdPrefix = transactionIdPrefix;
 	}
 
+	/**
+	 * Set the maximum time to wait when closing a producer; default 5 seconds.
+	 * @param closeTimeout the close timeout.
+	 * @since 1.3.11
```

Comment:
The `Duration` is a Java 8 feature. You may consider to provide a separate PR for Java 6 based branch

Reply:
Right; they have `close(long, TimeUnit)` there - I should fix these `@since`.

Reply:
Actually, even our 2.2.x only has that; so this is `@since 2.3`; I need think some more about how to support forward kafka-clients in those old versions.

Reply:
So, merging without back-porting?